crates/uv/tests/it/python_pin.rs

```diff
@@ -198,6 +200,131 @@ fn python_pin() {
     }
 }
 
+#[test]
+fn python_pin_global_if_no_local() -> Result<()> {
+    let context: TestContext = TestContext::new_with_versions(&["3.11", "3.12"]);
+    let uv = context.user_config_dir.child("uv");
+    uv.create_dir_all()?;
+
+    // // Without arguments, we attempt to read the current pin (which does not exist yet)
+    uv_snapshot!(context.filters(), context.python_pin(), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: No pinned Python version found
+    "###);
+
+    // Given an argument, we globally pin to that version
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.11").arg("--global"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Pinned `[UV_USER_CONFIG_DIR]/.python-version` to `3.11`
+
+    ----- stderr -----
+    ");
+
+    // If no local pin, use global.
+    uv_snapshot!(context.filters(), context.python_pin(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    3.11
+
+    ----- stderr -----
+    "###);
+
+    Ok(())
+}
+
+#[test]
+fn python_pin_global_use_local_if_available() -> Result<()> {
```

Comment:
I'd still just recommend combining this test with the one above and calling it `python_pin_global` since we tend to discuss the specific behavior we are testing in a comment above each invocation. This isn't blocking feedback though